src/Propulsion.CosmosStore/ChangeFeedProcessor.fs

```diff
@@ -114,44 +59,64 @@ type ChangeFeedProcessor =
-
-        let builder =
-            let feedProcessorOptions =
-                ChangeFeedProcessorOptions(
-                    StartFromBeginning=not (defaultArg startFromTail false),
-                    LeaseAcquireInterval=leaseAcquireInterval, LeaseExpirationInterval=leaseTtl, LeaseRenewInterval=leaseRenewInterval,
-                    FeedPollDelay=feedPollDelay,
-                    LeasePrefix=leasePrefix + ":")
-            // As of CFP 2.2.5, the default behavior does not afford any useful characteristics when the processing is erroring:-
-            // a) progress gets written regardless of whether the handler completes with an Exception or not
-            // b) no retries happen while the processing is online
-            // ... as a result the checkpointing logic is turned off.
-            // NB for lag reporting to work correctly, it is of course still important that the writing take place, and that it be written via the CFP lib
-            feedProcessorOptions.CheckpointFrequency.ExplicitCheckpoint <- true
-            // Max Items is not emphasized as a control mechanism as it can only be used meaningfully when events are highly regular in size
-            maxDocuments |> Option.iter (fun mi -> feedProcessorOptions.MaxItemCount <- Nullable mi)
-            let mkD cid (dc : DocumentClient) =
-                DocumentCollectionInfo(Uri=dc.ServiceEndpoint,ConnectionPolicy=dc.ConnectionPolicy,DatabaseName=cid.database,CollectionName=cid.container)
-            ChangeFeedProcessorBuilder()
-                .WithHostName(leaseOwnerId)
-                .WithFeedCollection(mkD source client)
-                .WithLeaseCollection(mkD aux (defaultArg auxClient client))
-                .WithFeedDocumentClient(client)
-                .WithLeaseDocumentClient(defaultArg auxClient client)
-                .WithProcessorOptions(feedProcessorOptions)
+        log.Information("ChangeFeed {processorName} Lease acquire {leaseAcquireIntervalS:n0}s ttl {ttlS:n0}s renew {renewS:n0}s feedPollDelay {feedPollDelayS:n0}s",
+            processorName, s leaseAcquireInterval, s leaseTtl, s leaseRenewInterval, s feedPollDelay)
+        let processorName_ =  processorName + ":"
+        let leaseTokenToPartitionId (leaseToken : string) = int (leaseToken.Trim[|'"'|])
+        let processor =
+            let handler // : Container.ChangeFeedHandlerWithManualCheckpoint
+                    (context : ChangeFeedProcessorContext)
+                    (changes : IReadOnlyCollection<Newtonsoft.Json.Linq.JObject>)
+                    (tryCheckpointAsync : Func<System.Threading.Tasks.Task<struct (bool*exn)>>)
+                    _ct = async {
+                let checkpoint = async {
+                    match! tryCheckpointAsync.Invoke() |> Async.AwaitTaskCorrect with
+                    | true, _ -> return ()
+                    | false, ex -> return! Async.Raise ex } 
+                let unixEpoch = DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc)
```

Comment:
[`DateTime.UnixEpoch`](https://docs.microsoft.com/en-us/dotnet/api/system.datetime.unixepoch) passes `dotnet build build.proj -v n`.

Reply:
Applied, good idea. (That's what I get for hitting F12 in Rider and unthinkingly inlining the V2 impl!)